app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -34,7 +34,13 @@ def fetch_middleware_servers
             host_instance = find_host_by_bios_uuid(machine_id) ||
                             find_host_by_bios_uuid(alternate_machine_id(machine_id))
 
-            if host_instance
+            if server[:properties].try { |prop| prop['In Container'] } == 'true'
+              container = Container.find_by(:name => machine_id)
```

Comment:
I'm not sure about this, but this is the only way I have to x-linking the container with the hawkular agent, but it seems like we need to change this, WDYT @pilhuhn